octobot_trading/exchanges/util/parser/orders_parser.py

```diff
@@ -0,0 +1,743 @@
+        self._parse_price(price)  # parse after type, status
+        self._parse_filled_price()  # parse after price
+        self._parse_average_price()
+        self._parse_amount(quantity)
+        self._parse_remaining()  # parse after amount and status
+        self._parse_filled_amount()  # parse after amount and remaining
+        self._parse_cost()
+        self._parse_reduce_only()  # parse after type
+        self._parse_tag()
+        self._parse_fees()
+        # self._parse_datetime(timestamp)  # remove? is it used?
+        # self._parse_last_trade_timestamp()  # remove? is it used?
+        # self._parse_quantity_currency()  # remove? is it used?
+
+        if check_completeness:
+            await self._fetch_if_missing()
+            self._create_debugging_report_for_record()
+
+        return self.formatted_record
+
+    def _parse_status(self, missing_status_value):
+        def missing_status():
+            if missing_status_value:
+                return missing_status_value
+            self._log_missing(OrderCols.STATUS.value, OrderCols.STATUS.value)
+
+        def try_cryptofeed_order_status(raw_order_status):
+            if raw_order_status == cryptofeed_constants.OPEN:
+                return OrderStatus.OPEN.value
+            elif raw_order_status == cryptofeed_constants.PENDING:
+                return OrderStatus.OPEN.value
+            elif raw_order_status == cryptofeed_constants.FILLED:
+                return OrderStatus.FILLED.value
+            elif raw_order_status == cryptofeed_constants.PARTIAL:
+                return OrderStatus.PARTIALLY_FILLED.value
+            elif raw_order_status == cryptofeed_constants.CANCELLED:
+                return OrderStatus.CANCELED.value
+            elif raw_order_status == cryptofeed_constants.UNFILLED:
+                return OrderStatus.OPEN.value
+            elif raw_order_status == cryptofeed_constants.EXPIRED:
+                return OrderStatus.EXPIRED.value
+            elif raw_order_status == cryptofeed_constants.FAILED:
+                return OrderStatus.REJECTED.value
+            elif raw_order_status == cryptofeed_constants.SUBMITTING:
+                return OrderStatus.PENDING_CREATION.value
+            elif raw_order_status == cryptofeed_constants.CANCELLING:
+                return OrderStatus.PENDING_CANCEL.value
+            elif raw_order_status == cryptofeed_constants.CLOSED:
+                return OrderStatus.CLOSED.value
+            elif raw_order_status == cryptofeed_constants.SUSPENDED:
+                pass  # todo is it canceled?
+
+        def found_status(raw_status):
+            raw_status = self.exchange.connector.client.parse_order_status(
+                raw_status
+            )  # isn't it already parsed through that ccxt side?
+            try:
+                return OrderStatus(raw_status).value
+            except ValueError:
+                if order_status := try_cryptofeed_order_status(raw_status):
```

Comment:
I think cryptofeed parsers and ccxt parsers should not be mixed as they are so different, we can't risk side effects between those

Reply:
but the naming sounds pretty bullet proof. we could copy it into octobot, so it'll break if they change it